packages/status-js/src/request-client/request-client.ts

```diff
@@ -0,0 +1,322 @@
+import { bytesToHex } from 'ethereum-cryptography/utils'
+import { Protocols } from 'js-waku'
+import { createLightNode } from 'js-waku/lib/create_waku'
+import { PeerDiscoveryStaticPeers } from 'js-waku/lib/peer_discovery_static_list'
+import { waitForRemotePeer } from 'js-waku/lib/wait_for_remote_peer'
+import { SymDecoder } from 'js-waku/lib/waku_message/version_1'
+
+import { peers } from '../consts/peers'
+import {
+  ApplicationMetadataMessage,
+  ApplicationMetadataMessage_Type,
+} from '../protos/application-metadata-message_pb'
+import { CommunityDescription } from '../protos/communities_pb'
+import { ProtocolMessage } from '../protos/protocol-message_pb'
+import { ContactCodeAdvertisement } from '../protos/push-notifications_pb'
+import { compressPublicKey } from '../utils/compress-public-key'
+import { generateKeyFromPassword } from '../utils/generate-key-from-password'
+import { idToContentTopic } from '../utils/id-to-content-topic'
+import { isClockValid } from '../utils/is-clock-valid'
+import { payloadToId } from '../utils/payload-to-id'
+import { recoverPublicKey } from '../utils/recover-public-key'
+import { mapChannel } from './map-channel'
+import { mapCommunity } from './map-community'
+import { mapUser } from './map-user'
+
+import type { ChannelInfo } from './map-channel'
+import type { CommunityInfo } from './map-community'
+import type { UserInfo } from './map-user'
+import type { WakuLight } from 'js-waku/lib/interfaces'
+import type { MessageV1 as WakuMessage } from 'js-waku/lib/waku_message/version_1'
+
+export interface RequestClientOptions {
+  environment?: 'production' | 'test'
+}
+
+class RequestClient {
+  public waku: WakuLight
+  /** Cache. */
+  public readonly wakuMessages: Set<string>
+
+  constructor(waku: WakuLight) {
+    this.waku = waku
+    this.wakuMessages = new Set()
+  }
+
+  static async start(options: RequestClientOptions): Promise<RequestClient> {
+    const { environment = 'production' } = options
+
+    let waku: WakuLight | undefined
+    let client: RequestClient | undefined
+
+    try {
+      // Waku
+      waku = await createLightNode({
+        defaultBootstrap: false,
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        emitSelf: true,
+        pingKeepAlive: 0,
+        relayKeepAlive: 0,
+        libp2p: {
+          peerDiscovery: [
+            new PeerDiscoveryStaticPeers(peers[environment], { maxPeers: 1 }),
+          ],
+        },
+      })
+      await waku.start()
+      await waitForRemotePeer(waku, [Protocols.Store], 10 * 1000)
+
+      client = new RequestClient(waku)
+    } catch (error) {
+      if (client) {
+        await client.stop()
```

Comment:
I don't see how this can happen. 🤔

Reply:
True. Nothing's happing after the client assignment, result of copy and pasting and removing `await client.community.start()`.

Addressed at https://github.com/status-im/status-web/pull/342/commits/341170bd979898391d8a6276003e7888281a6b5c.